lib/consumer.js

```diff
@@ -116,56 +89,48 @@ Consumer.prototype.getRegistrarAddressOnRegistry = function() {
  */
 Consumer.prototype.getThing = function(identity) {
   var self = this;
-  return new Promise(function(fulfill, reject) {
-    var encodedIdentity = OrUtils.urn.packer.encodeAndChunk(OrUtils.common.compressAll(identity));
-    self.registry.getThing.call(encodedIdentity, {from: self.address}, function(err, data) {
-      if (err) {
-        console.error(err);
-        reject(err);
-        return;
-      }
-
-      var encodedIdentities = data[0];
-      if (encodedIdentities.length == 0) {
-        reject('Thing is not existing');
-        return;
-      }
+  return Promise.try(function() {
+    return OrUtils.urn.packer.encodeAndChunk(OrUtils.common.compressAll(identity));
+  }).then(function(encodedIdentity) {
+    return self.registry.getThingAsync(encodedIdentity, {from: self.address});
+  }).then(function(data) {
```

Comment:
Similiar style comment here. If you use arrow functions you can get rid of the `self` declaration at the top. Whether or not these are one line functions is entirely up to you. Something like
```
return Promise.try(() => OrUtils.urnpacker.encodeAndCHunc(OrbUtils.common.compressAll(identity)))
.then(id => this.registry.getThingsAsync(id, {from: this.address})
.then(data =>
  ...
```